src/EditorFeatures/CSharpTest/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -0,0 +1,1632 @@
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private const string [|_goo|] = """";
+
+    [System.Obsolete(_goo)]
+    public void M() { }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task MethodIsInvoked()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|M1|] => 0
+    public int M2() => M1();
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task MethodIsAddressTaken()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|M1|] => 0
+    public void M2()
+    {
+        System.Func<int> m1 = M1;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task PropertyIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|P|] => 0;
+    public int M() => P;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task IndexerIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|this|][int x] { get { return 0; } set { } }
+    public int M(int x) => this[x];
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|e|];
+    public EventHandler P => e;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsSubscribed()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|e|];
+    public void M()
+    {
+        e += MyHandler;
+    }
+
+    static void MyHandler(object sender, EventArgs e)
+    {
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsRaised()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|_eventHandler|];
+
+    public void RaiseEvent(EventArgs e)
+    {
+        _eventHandler(this, e);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldInNameOf()
+        {
+            await TestDiagnosticsAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    private string _goo2 = nameof(_goo);
+}",
+    expected: Diagnostic("IDE0052", "_goo").WithLocation(3, 17));
```

Comment:
why do you need the location?  Isn't htat implied by the selected span?  it makes the test seem more redundant, and it is something to have to maintain if the test body is tweaked.

Reply:
This is the baseline generated when the test fails, and I just copy pasted the expected baseline.

---
In reply to: [213408069](https://github.com/dotnet/roslyn/pull/29511#discussion_r213408069) [](ancestors = 213408069)

Reply:
Ick.  I hate it.  Can you write a littel regex that just goes and removes these :)

Reply:
I can do so but given this is the format used by the underling test helper in `TestBase`, it is much easier to maintain, look at test failure diffs, write new tests and debug. Also note that squiggle text may not always match the annotation span.

---
In reply to: [213424411](https://github.com/dotnet/roslyn/pull/29511#discussion_r213424411) [](ancestors = 213424411)

Reply:
I have added logic in the added test helper to allow tests to specify diagnostic span/location to verify just based on annotation.

---
In reply to: [213476163](https://github.com/dotnet/roslyn/pull/29511#discussion_r213476163) [](ancestors = 213476163)